api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -48,6 +48,20 @@ const (
 	ManagedControlPlaneOutboundTypeUserDefinedRouting ManagedControlPlaneOutboundType = "userDefinedRouting"
 )
 
+// ManagedControlPlaneIdentityType enumerates the values for managed control plane identity type.
+type ManagedControlPlaneIdentityType string
```

Comment:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go#L520-L531 can we just reuse these existing consts?

Reply:
I'd prefer to keep these strongly typed and distinct, but I have inherited the actual string const values from the vm consts